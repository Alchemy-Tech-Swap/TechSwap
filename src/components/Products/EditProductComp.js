import React from 'react';

export default function EditProductComp({
  title,
  description,
  price,
  file,
  categories,
  condition,
  onStateChange,
  onSubmit,
}) {
  return (
    <div className="editProductForm">
      <form onSubmit={onSubmit}>
        <input
          name="title"
          id="title"
          type="text"
          value={title}
          onChange={(e) => onStateChange(e)}
        />
        <input
          name="price"
          id="price"
          type="text"
          placeholder="Price"
          value={price}
          onChange={(e) => onStateChange(e)}
        />
        <input
          name="category"
          id="categories"
          type="text"
          placeholder="Category"
          value={categories}
          onChange={(e) => onStateChange(e)}
        />
        <input
          name="condition"
          id="condition"
          type="text"
          placeholder="Condition"
          value={condition}
          onChange={(e) => onStateChange(e)}
        />
        <input
          name="description"
          id="description"
          type="text"
          placeholder="Description"
          value={description}
          onChange={(e) => onStateChange(e)}
        />
        <input
          name="file"
          id="image"
          type="file"
          placeholder="Image"
          onChange={(e) => onStateChange(e)}
        />

        <input type="submit" value="Submit" />
      </form>

      <button>Delete Product</button>
    </div>
  );
}
