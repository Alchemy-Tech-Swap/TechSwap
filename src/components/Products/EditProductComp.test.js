import EditProductComp from './EditProductComp';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';

test('Should render Edit Product Component', async () => {
  const { container } = render(
    <MemoryRouter>
      <EditProductComp
        products={{
          category: 'accessories',
          condition: 'Slightly used',
          description: 'High performance computer mouse',
          id: '34',
          onStateChange: 'ƒ onStateChange() {}',
          onSubmit: 'ƒ onSubmit() {}',
          price: 45,
          title: 'LogiTech MX Master 3',
        }}
      />
    </MemoryRouter>
  );

  await screen.findByText('Condition:');
  expect(container).toMatchSnapshot();
});
