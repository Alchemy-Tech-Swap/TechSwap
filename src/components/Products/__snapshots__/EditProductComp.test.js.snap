// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Edit Product Component 1`] = `
<div>
  <div
    class="editProductForm"
  >
    <form>
      <label>
        Title:
      </label>
      <input
        id="title"
        name="title"
        type="text"
        value=""
      />
      <label>
        Price:
      </label>
      <input
        id="price"
        name="price"
        placeholder="Price"
        type="text"
        value=""
      />
      <label>
        Category:
      </label>
      <input
        id="categories"
        name="category"
        placeholder="Category"
        type="text"
        value=""
      />
      <label>
        Condition:
      </label>
      <input
        id="condition"
        name="condition"
        placeholder="Condition"
        type="text"
        value=""
      />
      <label>
        Description:
      </label>
      <input
        id="description"
        name="description"
        placeholder="Description"
        type="text"
        value=""
      />
      <label>
        Image:
      </label>
      <input
        id="image"
        name="file"
        placeholder="Image"
        type="file"
      />
      <input
        class="submit"
        type="submit"
        value="Submit"
      />
    </form>
    <button
      class="delete"
    >
      Delete
    </button>
  </div>
</div>
`;
